Adapter/GADMAdapterMoPub.m

```diff
@@ -1,263 +0,0 @@
-      if (error) {
-          [self.connector adapter:self didFailAd:error];
-      }
-      else {
-          self.nativeAd = response;
-          self.nativeAd.delegate = self;
-      
-          if(options!=nil){
-    
-              for (GADNativeAdImageAdLoaderOptions *imageOptions in options) {
-                  if (![imageOptions isKindOfClass:[GADNativeAdImageAdLoaderOptions class]]) {
-                      continue;
-                  }
-                  
-                  // Load only image urls
-                  if(imageOptions.disableImageLoading)
-                  {
-                      _mediatedAd = [[MoPubAdapterMediatedNativeAd alloc] initWithMoPubNativeAd:self.nativeAd mappedImages:nil];
-                      [self.connector adapter:self didReceiveMediatedNativeAd:_mediatedAd];
-                  }
-              }
-          }
-        
-          [self loadNativeAdImages];
-
-      }
-
-  }];
-    
-  
-  NSLog(@"Requesting native ad from Sample Ad Network");
-}
-
-//Helper classes for downloading images
-
-- (void)loadNativeAdImages {
-    _imagesDictionary = [[NSMutableDictionary alloc] init];
-    // Load icon and cover image, and notify the  connector when completed.
-    NSURL *iconURL = [NSURL URLWithString:self.nativeAd.properties[kAdIconImageKey]];
-    NSURL *coverImageURL = [NSURL URLWithString:self.nativeAd.properties[kAdMainImageKey]];
-    [self loadCoverImageForURL:coverImageURL];
-    [self loadIconForURL:iconURL];
-}
-// Load main image for URL.
-- (void)loadCoverImageForURL:(NSURL *)coverImageURL {
-    __weak typeof(self) weakSelf = self;
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
-        __strong typeof(self) strongSelf = weakSelf;
-        NSData *imageData = [NSData dataWithContentsOfURL:coverImageURL];
-        UIImage *image = [UIImage imageWithData:imageData];
-        GADNativeAdImage *nativeAdImage = [[GADNativeAdImage alloc] initWithImage:image];
-        if (nativeAdImage) {
-            [strongSelf completedLoadingNativeAdImage:nativeAdImage imageKey:kAdMainImageKey];
-        }
-        });
-}
-// Load icon image for URL.
-- (void)loadIconForURL:(NSURL *)iconURL {
-    __weak typeof(self) weakSelf = self;
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
-        __strong typeof(self) strongSelf = weakSelf;
-        NSData *imageData = [NSData dataWithContentsOfURL:iconURL];
-        UIImage *image = [UIImage imageWithData:imageData];
-        GADNativeAdImage *nativeAdIcon = [[GADNativeAdImage alloc] initWithImage:image];
-        if (nativeAdIcon) {
-            [strongSelf completedLoadingNativeAdImage:nativeAdIcon imageKey:kAdIconImageKey];
-        }
-    });
-}
-
-- (void)completedLoadingNativeAdImage:(GADNativeAdImage *)image imageKey:(NSString *)imageKey {
-    
-    [_imagesDictionary setObject:image forKey:imageKey];
-   
-    if ([_imagesDictionary count] < 2) {
```

Comment:
Sorry for not mentioning it earlier. If you use - (void)addDownloadImageURLs:(NSArray *)imageURLs completionBlock:(MPImageDownloadQueueCompletionBlock)completionBlock and pass in both icon and cover images, the completion block is called only when both images are loaded.